### PR TITLE
Update to php 8 and migrated e2e test to php 8

### LINKED ISSUE #2923: Migrate e2e tests to php8
I think Deployer v7 will be the latest version that supports php7. And Deployer v8 will be only php8. 

Currently, e2e tests run on php7, we need to migrate tests to php8 eventually. 

@bsacharski do you think you can contribute more? ;)

One other thought is to also test the provision recipe. It even may be simpler as no apache will be needed as the provision recipe will install and configure everything himself.

### LINKED ISSUE #2958: Laravel 9 support
With laravel 9 update, nunomaduro/collision v6 is now a dependency, this projects needs symfony/console v5 but collision requires  console v6.

### PR DESCRIPTION
**Pull request by luigel**
- [x] Bug fix
- [ ] New feature?
- [x] BC breaks?
- [x] Tests added?
- [ ] Docs added?

fixes #2923 
fixes #2958

- dropped php 7.3. 
- migrated e2e tests to php 8
- Removed `#[\ReturnTypeWillChange]` and added type hints.
- Bump dependencies to only support php8 and symfony ^6

### CODE REVIEW

**Comment by antonmedv** [comment]
Awesome! Will check this later.

File: tests/e2e/recipe/laravel-boilerplate.php [hunk 1]
```diff
@@ -4,8 +4,8 @@
 require_once __DIR__ . '/hosts.php';
 require_once __DIR__ . '/../../../recipe/laravel.php';
 
-set('repository', 'https://github.com/deployphp/test-laravel.git');
-set('branch', 'main');
+set('repository', 'https://github.com/laravel/laravel.git');
+set('branch', 'master');
```
Comment (luigel) [inline tests/e2e/recipe/laravel-boilerplate.php:8]: When this https://github.com/deployphp/test-laravel/pull/1 PR is merged and created new branch named `laravel-9` apply this suggestion

```suggestion
set('repository', 'https://github.com/deployphp/test-laravel.git');
set('branch', 'laravel-9');
```

**Comment by antonmedv** [comment]
Let’s not update laravel to latest right now.

**Comment by antonmedv** [comment]
Also, I’m going to create v8 branch. Let’s add it there.

**Comment by luigel** [comment]
@antonmedv Sure. 😀

**Comment by chrispage1** [comment]
Would be great to see this implemented now Laravel 9 is out. I've upgraded to PHP 8.1 as standard on my machine and get lots of deprecation notices

**Comment by jasperf** [comment]
We are in the process of upgrading to Laravel 9 and see that dev-master (Deployer 7) is not working yet. Any updates on this pull request to remedy this?

**Comment by antonmedv** [comment]
Awesome work! Thanks!

**Pull request merged**